### DISCUSSION TITLE
Insert Jira issue data into the note

### DISCUSSION THREAD

**Discussion by grazor** [SilentVoid13/Templater#331] [Templates Showcase]
The [Gist](https://gist.github.com/grazor/6f482c459a4079fde92463388c60bc57).

It requires [go-jira](https://github.com/andygrunwald/go-jira) to be installed and configured, and user function `jira` to be described as `jira view -t debug $id`. The script extracts jira issue id from note file name and inserts its data to the beginning of the note as:

```
---
issue: ${y.key}
url: ${issueURL}
priority: ${y.fields.priority.name}
reporter: ${y.fields.reporter.displayName} (@${y.fields.reporter.name})
assignee: ${y.fields.assignee.displayName} (@${y.fields.assignee.name})
created_at: ${created}
---

# [${y.key}](${issueURL}) ${y.fields.summary}

${toM(y.fields.description)}
```

Template `sprint` creates sprint note with references to open jira issues (requires `jira_tasks` command to run `jira list -t debug -a <username>`) as follows:

```
---
starts_at: ${start}
ends_at: ${end}
issues: ${issues.length}
tags:
    - work
    - sprint
---

# Issues

* [[${i.id}]] ${i.description} ([jira](${i.url}))
* [[${i.id}]] ${i.description} ([jira](${i.url}))
....
```

**Comment by slaughtr** [comment]
Just commenting here for visibility in case anyone comes across this, I added a comment to the gist outlining how to change this workflow to prompt the user for a JIRA ticket ID so you can easily insert the info into existing notes.

**Reply by MaksimZinovev** [reply]
Awesome. I wish there would be such template for Azure Devops 🙂
